File: socialuni-server/socialuni-user-sdk/src/main/java/com/socialuni/social/user/sdk/model/factory/SocialuniContentUserROFactory.java
package com.socialuni.social.user.sdk.model.factory;

import com.socialuni.social.user.sdk.constant.UserIdentityAuthStatus;
import com.socialuni.social.user.sdk.logic.manage.SocialuniUserFollowManage;
import com.socialuni.social.common.api.model.user.SocialuniContentUserRO;
import com.socialuni.social.common.sdk.dao.DO.SocialuniUserDo;
import com.socialuni.social.user.sdk.model.DO.SocialUserIdentityAuthDO;
import com.socialuni.social.user.sdk.repository.SocialUserIdentityAuthRepository;
import org.springframework.stereotype.Component;

import javax.annotation.Resource;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class SocialuniContentUserROFactory {
    static SocialuniUserFollowManage followManage;

    static SocialUserIdentityAuthRepository socialUserIdentityAuthRepository;

    @Resource
    public void setSocialUserIdentityAuthRepository(SocialUserIdentityAuthRepository socialUserIdentityAuthRepository) {
        SocialuniContentUserROFactory.socialUserIdentityAuthRepository = socialUserIdentityAuthRepository;
    }

    @Resource
    public void setFollowManage(SocialuniUserFollowManage followManage) {
        SocialuniContentUserROFactory.followManage = followManage;
    }

    public static SocialuniContentUserRO newContentUserRO(SocialuniUserDo user, SocialuniUserDo mineUser) {
        SocialuniContentUserRO userRO = new SocialuniContentUserRO(SocialuniUserROFactory.getUserRO(user, mineUser));

        userRO.setHasFollowed(false);
        userRO.setHasBeFollowed(false);
        userRO.setIdentityAuth(false);

        if (mineUser != null && user.getUnionId().equals(mineUser.getUnionId())) {
            userRO.setIsMine(true);
        } else {
            userRO.setIsMine(false);
        }

        if (mineUser != null && !userRO.getIsMine()) {
            boolean hasFollowUser = followManage.userHasFollowBeUser(mineUser.getUnionId(), user.getUnionId());
            userRO.setHasFollowed(hasFollowUser);

            boolean hasBeFollowed = followManage.userHasFollowBeUser(user.getUnionId(), mineUser.getUnionId());
            userRO.setHasBeFollowed(hasBeFollowed);
        }

        SocialUserIdentityAuthDO socialUserIdentityAuthDO = socialUserIdentityAuthRepository.findFirstByUserId(user.getUnionId());
        if (socialUserIdentityAuthDO != null) {
            if (UserIdentityAuthStatus.authSuccessList.contains(socialUserIdentityAuthDO.getStatus())) {
                userRO.setIdentityAuth(true);
            }
        }
        return userRO;
    }

    public static List<SocialuniContentUserRO> toList(List<SocialuniUserDo> socialuniUserDos, SocialuniUserDo mineUser) {
        return socialuniUserDos.stream().map(item -> SocialuniContentUserROFactory.newContentUserRO(item, mineUser)).collect(Collectors.toList());
    }
}

File: socialuni-server/socialuni-follow-sdk/src/main/kotlin/com/socialuni/social/friend/facotry/SocialuniContentUserROFactory.kt
package com.socialuni.social.friend.facotry

import com.socialuni.social.common.api.model.user.SocialuniContentUserRO
import com.socialuni.social.common.sdk.dao.DO.SocialuniUserDo
import com.socialuni.social.user.sdk.constant.UserIdentityAuthStatus
import com.socialuni.social.user.sdk.logic.manage.SocialuniUserFollowManage
import com.socialuni.social.user.sdk.model.factory.SocialuniUserROFactory
import com.socialuni.social.user.sdk.repository.SocialUserIdentityAuthRepository
import org.springframework.stereotype.Component
import java.util.stream.Collectors
import javax.annotation.Resource

@Component
class SocialuniContentUserROFactory {
    @Resource
    fun setSocialUserIdentityAuthRepository(socialUserIdentityAuthRepository: SocialUserIdentityAuthRepository?) {
        Companion.socialUserIdentityAuthRepository = socialUserIdentityAuthRepository
    }

    @Resource
    fun setFollowManage(followManage: SocialuniUserFollowManage?) {
        Companion.followManage = followManage
    }

    companion object {
        var followManage: SocialuniUserFollowManage? = null

        var socialUserIdentityAuthRepository: SocialUserIdentityAuthRepository? = null

        fun newContentUserRO(user: SocialuniUserDo, mineUser: SocialuniUserDo?): SocialuniContentUserRO {
            val userRO = SocialuniContentUserRO(SocialuniUserROFactory.getUserRO(user, mineUser))

            userRO.hasFollowed = false
            userRO.hasBeFollowed = false
            userRO.identityAuth = false

            if (mineUser != null && user.unionId == mineUser.unionId) {
                userRO.isMine = true
            } else {
                userRO.isMine = false
            }

            if (mineUser != null && !userRO.isMine) {
                val hasFollowUser = followManage!!.userHasFollowBeUser(mineUser.unionId, user.unionId)
                userRO.hasFollowed = hasFollowUser

                val hasBeFollowed = followManage!!.userHasFollowBeUser(user.unionId, mineUser.unionId)
                userRO.hasBeFollowed = hasBeFollowed
            }

            val socialUserIdentityAuthDO = socialUserIdentityAuthRepository!!.findFirstByUserId(user.unionId)
            if (socialUserIdentityAuthDO != null) {
                if (UserIdentityAuthStatus.authSuccessList.contains(socialUserIdentityAuthDO.status)) {
                    userRO.identityAuth = true
                }
            }
            return userRO
        }

        fun toList(socialuniUserDos: List<SocialuniUserDo>, mineUser: SocialuniUserDo?): List<SocialuniContentUserRO> {
            return socialuniUserDos.stream().map { item: SocialuniUserDo -> newContentUserRO(item, mineUser) }
                .collect(Collectors.toList())
        }
    }
}
